Meadowlark-js/packages/meadowlark-core/src/handler/Query.ts

```diff
@@ -54,9 +55,15 @@ export async function query(frontendRequest: FrontendRequest): Promise<FrontendR
 
   writeDebugStatusToLog(moduleName, frontendRequest, 'query', 200);
   const body = JSON.stringify(documents);
+
+  const headers = {
+    ...frontendRequest.middleware.headerMetadata,
+    [TOTAL_COUNT_HEADER_NAME]: result.totalCount?.toString() ?? '0',
+  };
```

Comment:
There's one "problem" with this: the ODS/API only reports the total-count header if you use the query string `totalCount=true`. We're reporting in all cases. That said, it doesn't really matter. We didn't make any extra effort to get this information from OpenSearch, so let's just include it. So I'm fine with this.